### DISCUSSION TITLE
How to set bash formatter enable

### DISCUSSION THREAD

**Discussion by xbunax** [LazyVim/LazyVim#3133] [Q&A]
I use `mason` to install formatter `shfmt`, but I'm unable to use this and LazyVim has warning `no formatter available` like this
<img width="437" alt="image" src="https://github.com/LazyVim/LazyVim/assets/50800571/de0f93c9-e6d3-4e18-a143-0cc9886f0e77">

then I config the `conform.lua` to this 
```lua
return {
  "stevearc/conform.nvim",
  opts = {
    formatters_by_ft = {
      sh = { "shfmt" },
    },
    formatters = {
      shfmt = {
        prepend_args = { "-i", "2", "-ci", "-bn" },
      },
    },
  },
}
```
but the problem is still. I try to use injected format, still unable to use
<img width="1138" alt="image" src="https://github.com/LazyVim/LazyVim/assets/50800571/10d199be-7c2f-4788-86df-5d81f8aa45c3">

**Comment by rubiin** [comment]
check what filetype is set for the file you are trying to format . If ifs different , then you have to set it to sh
try `:echo &filetype` to get file type for file you are trying to format

**Reply by xbunax** [reply]
Thanks ,It's work for me.

**Reply by rubiin** [reply]
can you mark the comment as answer as well for future users with similar issue :)

**Reply by Wikiwix** [reply]
I mean that is only half the answer:
`bash` can be a valid filetype and there is a substantial difference between POSIX `sh` and `bash`'s features.

A local fix (while keeping the `bash` filetype) is to add

```lua
  opts = {
    formatters_by_ft = {
      bash = { "shfmt" },
    },
```

to the conform spec.
